### PR TITLE
Add Google Directory User List API Endpoint

### PR DESCRIPTION
**Pull request by TimCreasman**
Integrated the Google Directory User API with an API endpoint on the server-side of CheckIT. 
The information is pulled down based on a company domain and filtered to only retrieve the name and email of each user. This will be used in conjunction with #16 to provide a way for an admin to check out a device for any user within a company's user list, even if that user is not within the CheckIT database.

### CODE REVIEW

File: server/services/authService.js [hunk 1]
```diff
@@ -14,19 +14,28 @@ var authUtil = require('../util/authUtil.js');
 var errors = require('../util/errors.js'),
     mongoose = require('mongoose'),
     User = mongoose.model('User');
-var url, _logger, googleClientSecret, googleClientId, companyDomains;
+var google = require('googleapis');
+var path = require('path');
+var url, _logger, serviceAccount, serviceScopes, appAccountEmail,
```
Comment (tinoweiss) [inline server/services/authService.js]: Can this be updated to not be 3 separate lines (19-21) of variable declarations?

File: server/services/authService.js [hunk 2]
```diff
@@ -14,19 +14,28 @@ var authUtil = require('../util/authUtil.js');
...
+companyDomains, 
+serviceAccountEmail, serviceAccountKeyFile, appAccountEmail;
```
Comment (tinoweiss) [inline server/services/authService.js]: "appAccountEmail" seems to be defined twice

File: server/swagger/v1-api/definitions/GoogleUsers.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+items:
+  properties:
+    user:
+      properties:
+        email: {description: the email associated with this user, type: string}
+        name: {description: the full name of the user for displaying, type: string}
```
Comment (tinoweiss) [inline server/swagger/v1-api/definitions/GoogleUsers.yaml]: Is the name field an object or string? I remember seeing full name, first name, and last name fields. Be sure to explain that it includes all 3 if so.

File: server/services/authService.js (hunk 1 above)
Comment (TimCreasman) [inline server/services/authService.js]: Is two lines fine? One line is too long.

Comment (tinoweiss) [inline server/services/authService.js]: Yeah, two lines is fine. It looked odd to just have the companyDomains variable on a separate line.

**Pull request closed without merging**